src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,317 @@
+to obtain the corresponding image `Blob`. As before, this code is not tied
+to images, and will work with other future file types.
+
+```js
+async function getClipboardContents() {
+  try {
+    const clipboardItems = await navigator.clipboard.read();
+    for (const clipboardItem of clipboardItems) {
+      try {
+        for (const type of clipboardItem.types) {
+          const blob = await clipboardItem.getType(type);
+          console.log(URL.createObjectURL(blob));
+        }
+      } catch (e) {
+        console.error(e, e.message);
+      }
+    }
+  } catch (e) {
+    console.error(e, e.message);
+  }
+}
+```
+
+### Custom paste handler {: #custom-paste-handler }
+
+To dynamically handle paste events, listen for the `paste` event, call
+[`preventDefault()`][prevent-default] to prevent the default behavior
+in favor of your own logic, then use the code above to read the contents from
+the clipboard, and handle it in whatever way your app needs.
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  getClipboardContents();
+});
+```
+
+### Custom copy handler {: #custom-copy-handler }
+
+The [`copy` event][copy-event] includes a [`clipboardData`][clipboard-data]
+property with the items already in the right format, eliminating the need to
+manually create a blob. As before, don't forget to call `preventDefault()`.
+
+```js
+document.addEventListener('copy', async (e) => {
+  e.preventDefault();
+  try {
+    for (const item of e.clipboardData.items) {
+      await navigator.clipboard.write([
+        new ClipboardItem(Object.defineProperty({}, item.type, {
+          value: item,
+          enumerable: true
+        }))
+      ]);
+    }
+    console.log('Image copied.');
+  } catch(e) {
+    console.error(e, e.message);
+  }
+});
+```
+
+### Demo
+
+<iframe style="border: solid 1px;" width="100%" height="500"
+  src="https://jsfiddle.net/0794oysr/2/embedded/result,js,html,css"
+  allowfullscreen="allowfullscreen" frameborder="0">
+</iframe>
+
+### Security
+
+Opening up the Asynchronous Clipboard API for images comes with certain
+[risks](https://w3c.github.io/clipboard-apis/#security) that need to be
+carefully evaluated. One new challenge are so-called
+[image compression bombs](https://bomb.codes/bombs#images), that is, image
+files that appear to be innocent, but—once decompressed—turn out to be huge.
+Even more serious than large images are specifically crafted malicious images
+that are designed to exploit known vulnerabilities in the native operating
+system. This is why we can't just copy the image directly to the native
+clipboard, and why in Chrome we require that the image be transcoded.
+
+The [specification](https://w3c.github.io/clipboard-apis/#image-transcode)
+therefore also explicitly mentions transcoding as a mitigation method:
+
+> "To prevent malicious image data from being placed on the clipboard, the
+> image data may be transcoded to produce a safe version of the image."
```

Comment:
@robdodson this blockquote doesn't appear to be styled pretty, do you need to add some site styles?

Reply:
Yeah I think you have to put it in `<blockquote class="w-blockquote">`.

Which is kinda silly. We should make it so it just styles the blockquote element generically.

Reply:
I created #1650 to remind Rob to do that but he might not get to that soon, so for this one @petele I think the fast path is to just convert to HTML.

Reply:
Changed.